cookbooks/aws-parallelcluster-config/spec/unit/resources/manage_raid_mount_unmount_spec.rb

```diff
@@ -0,0 +1,196 @@
+        is_expected.to create_mdadm("MY_RAID")
+          .with(raid_device: '/dev/md0')
+          .with(level: 0)
+          .with(metadata: raid_superblock_version)
+          .with(devices: %w(/dev/disk/by-ebs-volumeid/vol-0 /dev/disk/by-ebs-volumeid/vol-1))
+
+        block = chef_run.find_resource('ruby_block', 'sleeping_for_raid_block')
+        expect(block).to receive(:wait_for_block_dev).with('/dev/md0')
+        block.block.call
+        expect(block).to subscribe_to('mdadm[MY_RAID]').on(:run).immediately
+      end
+
+      it "Setup raid disk" do
+        is_expected.to nothing_execute("setup_raid_disk")
+          .with(command: "sudo mkfs.ext4 /dev/md0")
+
+        command = chef_run.execute("setup_raid_disk")
+        expect(command).to subscribe_to('ruby_block[sleeping_for_raid_block]').on(:run).immediately
+      end
+
+      it "Create Raid config" do
+        is_expected.to nothing_execute("create_raid_config")
+          .with(command: "sudo mdadm --detail --scan | sudo tee -a /etc/mdadm.conf")
+
+        command = chef_run.execute("create_raid_config")
+        expect(command).to subscribe_to('execute[setup_raid_disk]').on(:run).immediately
+      end
+
+      it "Create the shared dir" do
+        is_expected.to create_directory('/raid_shared_dir')
+          .with(owner: 'root')
+          .with(group: 'root')
+          .with(mode: '1777')
+        # recursive do not work
+        # .with(recursive: true)
+      end
+
+      it "Mount and enable the shared dir" do
+        is_expected.to mount_mount('/raid_shared_dir')
+          .with(device: "/dev/md0")
+          .with(fstype: "ext4")
+          .with(options: %w(defaults nofail _netdev))
+          .with(retries: 10)
+          .with(retry_delay: 6)
+
+        is_expected.to enable_mount('/raid_shared_dir')
+          .with(device: "/dev/md0")
+          .with(fstype: "ext4")
+          .with(options: %w(defaults nofail _netdev))
+          .with(retries: 10)
+          .with(retry_delay: 6)
+      end
+
+      it "Change permission for the shared dir" do
+        is_expected.to create_directory('/raid_shared_dir')
+          .with(owner: 'root')
+          .with(group: 'root')
+          .with(mode: '1777')
+      end
+    end
+  end
+end
+
+describe 'manage_raid:unmount' do
+  for_all_oses do |platform, version|
+    context "on #{platform}#{version}" do
+      cached(:venv_path) { 'venv' }
+      cached(:chef_run) do
+        runner = ChefSpec::Runner.new(
+          platform: platform, version: version,
+          step_into: ['manage_raid']
+        ) do |node|
+          node.override['cluster']['cookbook_virtualenv_path'] = venv_path
+        end
+        runner.converge_dsl do
+          manage_raid 'unmount' do
+            action :unmount
+            raid_vol_array "vol-0, vol-1"
+            raid_shared_dir "raid_shared_dir"
+            raid_type ' 0'
+          end
+        end
+      end
+
+      it "Unmount and disable shared dir" do
```

Comment:
Nitpick. Here and everywhere: in case of failure you'll see something like `manage_raid:unmount on centos7 Unmount and disable shared dir`. This is why it would be better to start with a lowercase letter and use 3rd person: `unmounts and disables shared dir` (even if it is not clear what disabling shared dir means).